src/wasm.h

```diff
@@ -961,6 +961,7 @@ class If : public SpecificExpression<Expression::IfId> {
 
   void finalize() {
     if (ifFalse) {
+      assert(ifTrue && ifFalse);
```

Comment:
First thing, thanks for this! :)

And a comment on this line: this should not be possible, ifTrue must always exist. So this assertion is valid, however, the problem is happening at an earlier stage, so we should also add another fix somewhere. Is it possible to get the input that led to this?